cmake/api_macros.cmake

```
@@ -191,6 +191,21 @@ macro(OPENDDS_TARGET_SOURCES target)
     list(APPEND _tao_options "-SS")
   endif()
 
+  list(LENGTH ${CMAKE_CXX_COMPILER} cxx_compiler_length)
```

```suggestion
  list(LENGTH CMAKE_CXX_COMPILER cxx_compiler_length)
```

I played around with this and I don't think it returns the right value otherwise:

```cmake
set(list1 "1")
list(LENGTH ${list1} len1)
message("List is ${len1} ${list1}") # List is 0 1
list(LENGTH list1 len1)
message("List is ${len1} ${list1}") # List is 1 1
```

---

Yeah, I was having some issues with it when doing testing where it would jump from 0 to 2 in length.